package/MDAnalysis/analysis/base.py

```diff
@@ -125,14 +125,192 @@
+
+        Parameters
+        ----------
+        func : Callable
+            function to be applied
+        computations : list
+            list of computations (single argument only) to apply the function to
+
+        Returns
+        -------
+        list
+            list of results of each computation
+
+        Raises
+        ------
+        ValueError
+            if none of the conditions for applying each computing method is met
+        """
+        options = {
+            self._compute_with_client: self.client is not None,
+            self._compute_with_dask: self.backend == "dask",
+            self._compute_with_multiprocessing: self.backend == "multiprocessing",
+            self._compute_with_local: self.backend == "local",
+        }
+        for map_, condition in options.items():
+            if condition:
+                return map_(func, computations)
+        raise ValueError(f"Backend or client is not set properly: {self.backend=}, {self.client=}")
+
+    def _compute_with_local(self, func: Callable, computations: list) -> list:
+        """Performs all computations within current process, without using any parallel approach.
+
+        Parameters
+        ----------
+        func : Callable
+            function to be executed
+        computations : list
+            list of arguments to the function (without unpacking)
+
+        Returns
+        -------
+        list
+            list of results
+        """
+        return [func(task) for task in computations]
+
+    def _compute_with_multiprocessing(self, func: Callable, computations: list) -> list:
+        """Performs all computations using multiprocessing.Pool parallelization.
+
+        Parameters
+        ----------
+        func : Callable
+            function to be executed
+        computations : list
+            list of arguments to the function (without unpacking)
+
+        Returns
+        -------
+        list
+            list of results
+        """
+        from multiprocessing import Pool
+
+        with Pool(processes=self.n_workers) as pool:
+            results = pool.map(func, computations)
+        return results
+
+    def _compute_with_dask(self, func: Callable, computations: list) -> list:
+        """Performs all computations using dask.compute(), which under the hood uses multiprocessing.Pool
+        but a different method for serialization (instead of default `pickle`), hence allowing for
+        potentially more functions to be executed in parallel.
+
+        Parameters
+        ----------
+        func : Callable
+            function to be executed
+        computations : list
+            list of arguments to the function (without unpacking)
+
+        Returns
+        -------
+        list
+            list of results
+        """
+        from dask.delayed import delayed
+        import dask
+        from multiprocessing import Pool
```

Comment:
```suggestion
```
?

Reply:
a legacy of passing `Pool` directly, before Yuxuan discovered the `chunksize=1` trick. Removed.